alt_e2eshark/onnx_tests/models/model.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2024 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+import os
+import numpy
+import onnx
+import torch
+from onnxruntime.tools.onnx_model_utils import make_dim_param_fixed, fix_output_shapes
+from e2e_testing import azutils
+from e2e_testing.framework import OnnxModelInfo
+from e2e_testing.registry import register_test
+from e2e_testing.storage import TestTensors
+from .protected_list import protected_models, models_with_postprocessing
+
+class AzureDownloadableModel(OnnxModelInfo):
+    def __init__(self, name: str, onnx_model_path: str, cache_dir: str):
```

Comment:
I think we should stipulate this to always look for an onnx_model in the onnx_model path, and make cache_dir an environment variable instead. Reduce the number of paths to keep track of. Bonus points if we can even make the name a part of that.

Reply:
Adding this to the todo list after merging. Having to specify the cache directory each time is kind of annoying, and figuring out a better way to deal with path management in the class structure is also something we should look into.